### ISSUE TITLE
RTX Failure: T1-init

### ISSUE THREAD

**Issue by maxhawkins** [pions/sctp#33]
### What did you do?
Added two PeerConnections and established Opus streams between them

### What did you expect?
No logs.

### What happened?
Pion printf logged "RTX Failure: T1-init". The audio stream works fine.

Should this happen normally or is it a bug?

**Comment by enobufs** [comment]
When that happened, was the data channel connection successful?

**Comment by maxhawkins** [comment]
I am not using data channels

**Comment by enobufs** [comment]
Hmm...  Then I guess, data channel is being opened without the app telling pion to do so might be the problem...

**Comment by maxhawkins** [comment]
pions/webrtc@d906c2b43b07

**Comment by maxhawkins** [comment]
Not sure if this is a separate issue: I'm also getting `Failed to retransmit init (nRtos=6): EOF` in the same application now.

**Comment by enobufs** [comment]
The "RTX Failure: T1-init" happens when all of 8-INIT chunk retransmissions have failed. You just saw the 6th retransmission failure on sending data to DTLS net.Conn. What I don't understand is why the datachannel handshake got started when your app is not using data channel. (smell of bug in datachannel..)

**Comment by enobufs** [comment]
I made a quick pion-pion connection test without data channel. From the SCTP log, it is trying to open a stream.

Although my code does not call CreateDataChannel(), I see m= line for DTLS/SCTP.  How does that get it the SDP??

```
$ PIONS_LOG_TRACE=sctp ./simple_dc
EP1 => EP2:
{"type":"offer","sdp":"v=0\r\no=- 393338590 1554187959 IN IP4 0.0.0.0\r\ns=-\r\nt=0 0\r\na=fingerprint:sha-256 2B:79:70:40:B2:AE:C9:C4:94:CA:81:32:4B:11:30:26:EC:DB:57:D5:4E:48:71:B1:52:93:17:62:58:C4:17:45\r\na=group:BUNDLE audio video data\r\nm=audio 9 UDP/TLS/RTP/SAVPF 111 9\r\nc=IN IP4 0.0.0.0\r\na=setup:actpass\r\na=mid:audio\r\na=ice-ufrag:hhgkESlHqcdXSyMl\r\na=ice-pwd:INlrzEelLDOoIkMDOyxNLDHejSKfPQRB\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:111 opus/48000/2\r\na=fmtp:111 minptime=10;useinbandfec=1\r\na=rtpmap:9 G722/8000\r\na=ssrc:680014774 cname:pion1\r\na=ssrc:680014774 msid:pion1 pion1\r\na=ssrc:680014774 mslabel:pion1\r\na=ssrc:680014774 label:pion1\r\na=sendrecv\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=end-of-candidates\r\na=setup:actpass\r\nm=video 9 UDP/TLS/RTP/SAVPF 96 100 98\r\nc=IN IP4 0.0.0.0\r\na=setup:actpass\r\na=mid:video\r\na=ice-ufrag:hhgkESlHqcdXSyMl\r\na=ice-pwd:INlrzEelLDOoIkMDOyxNLDHejSKfPQRB\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:96 VP8/90000\r\na=rtpmap:100 H264/90000\r\na=fmtp:100 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f\r\na=rtpmap:98 VP9/90000\r\na=recvonly\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=end-of-candidates\r\na=setup:actpass\r\nm=application 9 DTLS/SCTP 5000\r\nc=IN IP4 0.0.0.0\r\na=setup:actpass\r\na=mid:data\r\na=sendrecv\r\na=sctpmap:5000 webrtc-datachannel 1024\r\na=ice-ufrag:hhgkESlHqcdXSyMl\r\na=ice-pwd:INlrzEelLDOoIkMDOyxNLDHejSKfPQRB\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60810 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 54892 typ srflx raddr 10.0.0.135 rport 54892 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 54893 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 54893 generation 0\r\na=end-of-candidates\r\na=setup:actpass\r\n"}
EP2 => EP1:
{"type":"answer","sdp":"v=0\r\no=- 389034083 1554187959 IN IP4 0.0.0.0\r\ns=-\r\nt=0 0\r\na=fingerprint:sha-256 A9:D7:E5:A7:FC:7C:25:6B:56:33:3A:A7:08:58:07:D4:E0:58:90:92:59:0C:C9:36:84:6F:D9:EF:10:E4:18:1F\r\na=group:BUNDLE audio video data\r\nm=audio 9 UDP/TLS/RTP/SAVPF 111 9\r\nc=IN IP4 0.0.0.0\r\na=setup:active\r\na=mid:audio\r\na=ice-ufrag:ZUMANSaIuVyWprPB\r\na=ice-pwd:RFxKSSpGfYfqoofUoJMXgNSPttBNdUhj\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:111 opus/48000/2\r\na=fmtp:111 minptime=10;useinbandfec=1\r\na=rtpmap:9 G722/8000\r\na=ssrc:1839779657 cname:pion1\r\na=ssrc:1839779657 msid:pion1 pion1\r\na=ssrc:1839779657 mslabel:pion1\r\na=ssrc:1839779657 label:pion1\r\na=sendrecv\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=end-of-candidates\r\nm=video 9 UDP/TLS/RTP/SAVPF 96 100 98\r\nc=IN IP4 0.0.0.0\r\na=setup:active\r\na=mid:video\r\na=ice-ufrag:ZUMANSaIuVyWprPB\r\na=ice-pwd:RFxKSSpGfYfqoofUoJMXgNSPttBNdUhj\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:96 VP8/90000\r\na=rtpmap:100 H264/90000\r\na=fmtp:100 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f\r\na=rtpmap:98 VP9/90000\r\na=inactive\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=end-of-candidates\r\nm=application 9 DTLS/SCTP 5000\r\nc=IN IP4 0.0.0.0\r\na=setup:active\r\na=mid:data\r\na=sendrecv\r\na=sctpmap:5000 webrtc-datachannel 1024\r\na=ice-ufrag:ZUMANSaIuVyWprPB\r\na=ice-pwd:RFxKSSpGfYfqoofUoJMXgNSPttBNdUhj\r\na=candidate:foundation 1 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 2 udp 3776 10.0.0.135 60232 typ host generation 0\r\na=candidate:foundation 1 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 2 udp 3100 98.207.205.254 51517 typ srflx raddr 10.0.0.135 rport 51517 generation 0\r\na=candidate:foundation 1 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=candidate:foundation 2 udp 3100 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 51518 typ srflx raddr 2601:646:c200:93a4:d1cc:a5dc:7272:65c6 rport 51518 generation 0\r\na=end-of-candidates\r\n"}
sctp TRACE: 23:52:42.044923 association.go:220: updated cwnd=4380 ssthresh=0 inflight=0 (INI)
sctp TRACE: 23:52:42.044978 association.go:220: updated cwnd=4380 ssthresh=0 inflight=0 (INI)
sctp DEBUG: 23:52:42.044999 association.go:254: sending INIT
sctp DEBUG: 23:52:42.044980 association.go:254: sending INIT
sctp DEBUG: 23:52:42.045372 association.go:457: [0xc0003ae480] state change: 'Closed' => 'CookieWait'
sctp DEBUG: 23:52:42.045414 association.go:457: [0xc0001466c0] state change: 'Closed' => 'CookieWait'
sctp DEBUG: 23:52:42.045488 association.go:1517: [0xc0001466c0] chunkInit received in state 'CookieWait'
sctp DEBUG: 23:52:42.045514 association.go:1517: [0xc0003ae480] chunkInit received in state 'CookieWait'
sctp DEBUG: 23:52:42.045683 association.go:1543: [0xc0003ae480] chunkInitAck received in state 'CookieWait'
sctp DEBUG: 23:52:42.045720 association.go:526: initial rwnd=131072
sctp DEBUG: 23:52:42.045701 association.go:1543: [0xc0001466c0] chunkInitAck received in state 'CookieWait'
sctp TRACE: 23:52:42.045732 association.go:533: updated cwnd=4380 ssthresh=131072 inflight=0 (INI)
sctp DEBUG: 23:52:42.045762 association.go:526: initial rwnd=131072
sctp DEBUG: 23:52:42.045785 association.go:281: sending COOKIE-ECHO
sctp TRACE: 23:52:42.045791 association.go:533: updated cwnd=4380 ssthresh=131072 inflight=0 (INI)
sctp DEBUG: 23:52:42.045846 association.go:281: sending COOKIE-ECHO
sctp DEBUG: 23:52:42.045874 association.go:457: [0xc0003ae480] state change: 'CookieWait' => 'CookieEchoed'
sctp DEBUG: 23:52:42.045922 association.go:457: [0xc0001466c0] state change: 'CookieWait' => 'CookieEchoed'
sctp DEBUG: 23:52:42.045961 association.go:1580: [0xc0001466c0] chunkCookieEcho received in state 'CookieEchoed'
sctp DEBUG: 23:52:42.045977 association.go:457: [0xc0001466c0] state change: 'CookieEchoed' => 'Established'
sctp DEBUG: 23:52:42.046002 association.go:1580: [0xc0003ae480] chunkCookieEcho received in state 'CookieEchoed'
sctp DEBUG: 23:52:42.046104 association.go:457: [0xc0003ae480] state change: 'CookieEchoed' => 'Established'
sctp DEBUG: 23:52:42.046238 association.go:1589: [0xc0003ae480] chunkCookieAck received in state 'Established'
sctp DEBUG: 23:52:42.046240 association.go:1589: [0xc0001466c0] chunkCookieAck received in state 'Established'
```

**Comment by enobufs** [comment]
Sorry, I wasn't thinking properly last night. Unless pions support a renegotiation of m line, pions need to let SCTP/DTLS open regardless of users' intention to use data channel.

I believe there was a lot of packet loss when you saw those logs. `Failed to retransmit init (nRtos=6): EOF` though, underlying DTLS connection must have been closed for some reason.  Anyway, I believe you could ignore those if you are not using data channel. Those logs will be using the new logging facility once PR #26 lands.

**Comment by maxhawkins** [comment]
Got it. I'll just ignore these messages then. Thanks!

**Comment by enobufs** [comment]
#34 may also fix the errors you reported.

**Comment by maxhawkins** [comment]
I will update when it merges and let you know it that fixes things
